doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,226 @@
+## Multiple y cable vendor API design
+
+### Overview
+
+This document summarizes the approach taken to accommodate multiple
+Y cable vendors for supporting dual-ToR configurations in SONiC.
+All the design choices are presented
+such that vendors can implement the Y cable interface with ease and effeciency
+and all the feature requirements are met.
+
+
+Challenge: to provide an interface for Y cable to interact with PMON docker and HOST
+           which can be uniform across all vendors.
+
+### Directory Structure
+
+#### Background
+
+- We need to have an appropriate place where vendors can implement their modules/packages and provide Sonic with an abstraction layer where pmon and host can access the Y cable packages of all the vendors with ease
+   
+  - Currently the assumption is Y cable package implementation would be in python itself
+  - If a vendor provides shared library, that scenario would be considered accordingly
+  - Since sonic_y_cable (sonic-platform-common/sonic_y_cable) is already built as a package today (only for a single Y cable vendor with sonic-buildimage) vendors can also build packages along similar lines.
+
+#### Proposed Solution
+
+- Vendors can place their implementations in this format
```

Comment:
I think these examples should come after the "Vendor to Module mapping" section, because I think we should present a few examples
1. Vendor has a 1:1 mapping between model and Python file
2. Vendor has one Python file which supports all models
3. A combination of the above

Thus, I think we need to show both the directory structure and the mapping file together for each of these scenarios.

Reply:
fixed